src/client/App.jsx

```diff
@@ -1,56 +1,119 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import { Routes, Route, Link, useNavigate } from "react-router-dom";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
-  const [movies, setMovies] = useState([]);
+  const [movies, setMovies] = useState(null);
+  const [updateMovieList, setUpdateMovieList] = useState(0);
+
+  const navigate = useNavigate();
 
   useEffect(() => {
-    fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
-  }, []);
+    fetch(`${apiUrl}/movie`, {
```

Comment:
Extract client behaviour into its own module - so this might become instead

```js
client.get('/movie')
  .then(res => setMovies(res.data))
```

Reply:
Do you mean extracting the fetch request that is inside the useEffect?